fdbclient/SkipList.cpp

```diff
@@ -29,10 +29,10 @@
 #include "flow/Platform.h"
```

Comment:
Is this file copied or moved to fdbclient? And, while I do appreciate the formatting and documentations added here, these changes don't seem to be related to the purpose of this PR. Consider moving these changes to a separate PR to help the review-ability of this one.

Reply:
That's a fair point and this file is moved.

When doing my initial implementation, I was thinking using it on the client side and made the move. The same applies to the changes to Resolvers. Later I realized a much simpler implementation on top of the original one. Hopefully these refactoring is self-contained and can be separated.

Reply:
I separated SkipList changes to #3903